collect_app/src/main/java/org/odk/collect/android/javarosawrapper/XFormParser.kt

```diff
@@ -0,0 +1,41 @@
+package org.odk.collect.android.javarosawrapper
+
+import org.javarosa.xform.parse.XFormParser
+import org.kxml2.kdom.Document
+import org.kxml2.kdom.Element
+import org.xmlpull.v1.XmlPullParserException
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.InputStream
+import java.io.InputStreamReader
+
+/**
+ * This is a basic XML parser for forms, designed to read the content without
+ * the advanced parsing typically performed by JavaRosa. Currently, it provides
+ * simple functionality but can be extended to meet future requirements.
+ */
+object XFormParser {
+    @JvmStatic
+    @Throws(FileNotFoundException::class, XmlPullParserException::class)
+    fun parseXml(formFile: File): Document {
+        return parseXml(formFile.inputStream())
+    }
+
+    @JvmStatic
+    @Throws(XmlPullParserException::class)
+    fun parseXml(formInputStream: InputStream): Document {
+        return XFormParser.getXMLDocument(InputStreamReader(formInputStream))
```

Comment:
I'd not noticed `getXMLDocument`! Maybe for the moment we can just that directly everywhere and remove this file? I'm a little worried about the extension functions as they could be confusing when working with a `Document` that's not an XForm.

Reply:
Ok we can do that.